### PR TITLE
CB: Single quote strings in query rendering

### PR DESCRIPTION
**Pull request by drostron**
Fixes: #1751

### CODE REVIEW

File: couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala [hunk 1]
```diff
@@ -41,6 +41,8 @@ object RenderQuery {
   }
 
   val alg: AlgebraM[PlannerError \/ ?, N1QL, String] = {
+    case Data(s @ QData.Str(v)) =>
```
Comment (alissapajer) [inline couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala]: Looks like `s` isn't used below.

Comment (drostron) [inline couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala]: That's true — a leftover ready for ✂️.

File: couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala [hunk 2]
```diff
@@ -41,6 +41,8 @@ object RenderQuery {
   }
 
   val alg: AlgebraM[PlannerError \/ ?, N1QL, String] = {
+    case Data(QData.Str(v)) =>
+      ("'" ⊹ v.flatMap { case ''' => "''"; case v   => v.toString } ⊹ "'").right
```
Comment (alissapajer) [inline couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala]: A question about `case '''`: is this the intention to turn a single `'` into `''`?
```
scala> "'".flatMap { case ''' => "''"; case v => v.toString }
res1: String = ''

scala> "'ab'c".flatMap { case ''' => "''"; case v => v.toString }
res2: String = ''ab''c
```

Comment (alissapajer) [inline couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala]: Noting that we'd include an additional `"'"` on either end, returning `''''` and `'''ab''c'`, respectively.

Comment (drostron) [inline couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala]: Correct that is the intention and N1QL's way of quoting `'`s within single quoted strings.

Comment (alissapajer) [inline couchbase/src/main/scala/quasar/physical/couchbase/RenderQuery.scala]: Ok cool!

**Comment by alissapajer** [comment]
LGTM - will merge when green

**Pull request merged**